token-client/src/main/java/com/sap/cloud/security/xsuaa/mtls/X509Parser.java

```diff
@@ -0,0 +1,100 @@
+package com.sap.cloud.security.xsuaa.mtls;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Base64;
+
+public class X509Parser {
+	public static final Logger LOGGER = LoggerFactory.getLogger(X509Parser.class);
+
+	private X509Parser() {
+	}
+
+	/**
+	 * Generates X509 thumbprint from provided certificate as specified in JWT
+	 * Certificate Thumbprint Confirmation Method
+	 * 
+	 * @see <a href="https://tools.ietf.org/html/rfc8705#section-3.1"></a>
+	 * @param certificate
+	 *            Base64 encoded X509 certificate
+	 * @return Thumbprint of X509 certificate
+	 * @throws NoSuchAlgorithmException
+	 *             when a particular cryptographic algorithm is requested but is not
+	 *             available in the environment
+	 * @throws CertificateException is thrown if errors occur while decoding X509 certificate
+	 */
+	public static String getX509Thumbprint(String certificate) throws NoSuchAlgorithmException, CertificateException {
+		X509Certificate x509Certificate;
+
+		x509Certificate = decodeX509(certificate);
+		MessageDigest digest = MessageDigest.getInstance("SHA-256");
+		byte[] hashedX509 = digest.digest(x509Certificate.getEncoded());
+		String base64UrlEncoding = normalizeThumbprint(Base64.getUrlEncoder().encodeToString(hashedX509));
+		LOGGER.debug("Base64-url encoded thumbprint: {}", base64UrlEncoding);
+		return base64UrlEncoding;
+	}
+
+	public static String getJwtThumbprint(String base64cert) throws NoSuchAlgorithmException {
+		base64cert = base64cert
```

Comment:
still required? Wasn't normalization done by X509Certificate?

Reply:
this is the thumbprint generator that doesn't use X509Certificate class, possibly need to remove, or test the performance for both options and then decide

Reply:
I would keep only one: standard javax.security functionality would be my favorite